blockchain/chain_sync/src/sync.rs

```diff
@@ -830,23 +859,40 @@ mod tests {
         });
     }
 
+    fn dummy_header() -> BlockHeader {
+        BlockHeader::builder()
+            .miner_address(Address::new_id(1000).unwrap())
+            .messages(Cid::new_from_cbor(&[1, 2, 3], Blake2b256))
+            .message_receipts(Cid::new_from_cbor(&[1, 2, 3], Blake2b256))
+            .state_root(Cid::new_from_cbor(&[1, 2, 3], Blake2b256))
+            .build()
+            .unwrap()
+    }
     #[test]
     fn chainsync_constructor() {
-        let db = MemoryDB::default();
+        let db = Arc::new(MemoryDB::default());
+        let mut chain_store = ChainStore::new(db);
         let (local_sender, _test_receiver) = channel(20);
         let (_event_sender, event_receiver) = channel(20);
 
+        let gen_header = dummy_header();
+        chain_store.set_genesis(gen_header).unwrap();
         // Test just makes sure that the chain syncer can be created without using a live database or
         // p2p network (local channels to simulate network messages and responses)
-        let _chain_syncer = ChainSyncer::new(Arc::new(db), local_sender, event_receiver).unwrap();
+        let _chain_syncer =
+            ChainSyncer::new(chain_store, local_sender, event_receiver, None).unwrap();
     }
 
     #[test]
     fn sync_headers_reverse_given_tipsets_test() {
-        let db = MemoryDB::default();
+        let db = Arc::new(MemoryDB::default());
+        let mut chain_store = ChainStore::new(db);
+        let gen_header = dummy_header();
+        chain_store.set_genesis(gen_header.clone()).unwrap();
```

Comment:
Ok I guess I'll comment on each one instead lol
```suggestion
        chain_store.set_genesis(gen_header).unwrap();
```

Reply:
im sorry lol

Reply:
all good, just slid the all word in the last comment, could have said clearer